ts/Core/Series/Series.ts

```diff
@@ -4430,6 +4449,18 @@ class Series {
             series.isDirtyData = true;
         }
 
+        // If dataSorting was turned on, set the indexes
+        if (
+            options.dataSorting &&
+            options.dataSorting.enabled === true &&
+            (
+                !oldOptions.dataSorting ||
+                oldOptions.dataSorting.enabled === false
+            )
```

Comment:
Again, do we need all the conditions? Or would it be good enough to test for `'dataSorting' in options`? Sure, it would kick in when changing anything else in `dataSorting`, but does it matter?